onadata/libs/utils/viewer_tools.py

```diff
@@ -367,3 +366,27 @@ def get_enketo_preview_url(request, username, id_string, xform_pk=None):
             raise EnketoError(response['message'])
 
     return False
+
+
+def get_submission_url(request, username, id_string, xform_pk=None):
+    """Return submission url of the submission instance."""
+    enketo_url = settings.ENKETO_URL + "/api/v2/survey/single/once"
+    form_id = id_string
+    server_url = get_form_url(
+        request, username, settings.ENKETO_PROTOCOL, True, xform_pk=xform_pk)
+
+    url = '{}?server_url={}&form_id={}'.format(
+        enketo_url, server_url, form_id)
+
+    response = requests.get(url, auth=(settings.ENKETO_API_TOKEN, ''))
+
+    if response.status_code == 200:
+        data = response.json()
+        submission_url = data['single_url']
+        return submission_url
+    elif response.status_code == 400:
```

Comment:
Since we are raising the same exception with the same error message we probably should only have only the else clause. Check `enketo_url()` function, could be something that might need to be refactored into a function that can be used by both.